### PR TITLE
mypy: add coverage to tests

### PR DESCRIPTION
**Pull request by cjp256**
mypy.ini: set python version to 3.6
tests: minor fixups for mypy to run successfully
runtests: add mypy coverage of unit tests for static target

### CODE REVIEW

File: runtests.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ run_static_tests() {
     python3 -m flake8 .
 
     echo "Running mypy"
-    mypy --python-version 3.6 -p snapcraft
+    mypy --python-version 3.6 snapcraft tests
```
Comment (sergiusens) [inline runtests.sh]: you might be able to drop `--python-version 3.6`, I missed the change on mypy.ini (couldn't find where it was back then)

File: tests/fixture_setup/_unittests.py [hunk 2]
```diff
@@ -532,14 +532,14 @@ def __str__(self):
...
 
-    @property
-    def version(self):
-        return self._version
-
     @property
     def is_auto_removable(self):
         return self.marked_install and self.is_auto_installed
 
+    @property
```
Comment (sergiusens) [inline tests/fixture_setup/_unittests.py]: is there a reason for this change?

Comment (cjp256) [reply tests/fixture_setup/_unittests.py]: > is there a reason for this change?

Yeah, mypy was erroring out on the following setter.  It appeared to need the @property next to the setter.

Comment (sergiusens) [reply tests/fixture_setup/_unittests.py]: gotcha

**Comment by cjp256** [comment]
Should be good to go.

**Pull request merged**